redaxo/src/addons/backup/lib/cronjob.php

```diff
@@ -16,6 +16,12 @@ public function execute()
         $file = $filename;
         $dir = rex_backup::getDir() . '/';
         $ext = '.cronjob.sql';
+
+        $tables = rex_sql::factory()->getTables(rex::getTablePrefix());
```

Comment:
vorher gab es folgende Bedingungen, welche Tabellen exportiert werden sollen:

https://github.com/redaxo/redaxo/blob/68e674123cbd9615bdfea38b5b338cd30fd91f02/redaxo/src/addons/backup/lib/backup.php#L340-L349

nach dieser Änderung wird die Tabelle rex_user und tabellen die mit dem prefix `rex_tmp` beginnen, vermutlich im backup per default enthalten sein

Vorschlag:

man erzeuge eine neue methode, z.B. `rex_backup::getTables()` und packe dort die logik aus 
https://github.com/redaxo/redaxo/blob/68e674123cbd9615bdfea38b5b338cd30fd91f02/redaxo/src/addons/backup/lib/backup.php#L340-L349
rein.

Reply:
Ein sinnvoller Vorschlag, auf der anderen Seite entzieht man dem Entwickler die Kontrolle. Persönlich möchte ich auch mal die user-Tabelle ohne Umwege exportieren können oder sichern. Müsste man dann nicht noch ne zusätzliche Auswahl bereitstellen?

Reply:
> Persönlich möchte ich auch mal die user-Tabelle ohne Umwege exportieren können oder sichern. Müsste man dann nicht noch ne zusätzliche Auswahl bereitstellen?

ich denke das wäre ein separates issue.

ggf. macht es sinn eine checkbox anzubieten um rex_user doch mit drinnen zu haben (bzw. die rex_tmp_ sachen).

man müsste gregor nochmal fragen warum wir rex_user weglassen

Reply:
ich schätze, rex_user wurde weggelassen wegen des Hashs - da ein Redakteur ja so mit einer Rainbow-Table arbeiten konnte. Was aber nur unter md5 noch möglich war.

Vielleicht auch, weil sich ein Redakteur durch export/import so leichter zu einem Admin machen konnte?

Reply:
@gharlan kannst du was dazu sagen?

Reply:
Dass die Usertabelle bei Exporten ignoriert wird, ist ja schon seit Urzeiten so. Damals waren die Passwörter teils sogar noch Klartext.
Ob es jetzt noch gute Gründe gibt, die wegzulassen, weiß ich ehrlich gesagt nicht.

Ist es bei dem normalen Export (nicht über Cronjob) nicht so, dass alle Tabellen, außer rex_user vorausgewählt sind, und man so auch rex_user optional noch anhaken kann?
Kann man das hier nicht genauso machen?

Reply:
Ich würde es deswegen nicht so machen, weil mit jeder Addon-Installation neue Tabellen hinzukommen können, die auch in ein automatisches Backup gehören. Daher eine Blacklist und keine Whitelist.

Reply:
Guter Punkt! Das wäre in der Tat ärgerlich, wenn neue Tabellen dann fehlen würden.

Dann vielleicht wirklich mit der Checkbox für die Usertabelle?
Die tmp-Tabellen würde ich immer weglassen und keine Checkbox anbieten.

Oder doch das Verhalten ändern, wenn uns dreien keine guten Gründe einfallen, warum die Usertabelle ausgeschlossen werden sollte?

Reply:
Bzw. wenn es eine Auswahlbox für die Blacklist ist, kann dort nicht einfach default die User-Tabelle angehakt sein? Und man kann sie dann abwählen.

Reply:
> Oder doch das Verhalten ändern, wenn uns dreien keine guten Gründe einfallen, warum die Usertabelle ausgeschlossen werden sollte?

+1

Nicht nur, weil mein PR dann so bleibt, sondern, weil es schlüssig klingt.

Ein Szenario ist mir noch eingefallen: Ich bin Admin, im Backup gibt's den Admin nicht. Wenn ich jetzt das Backup importiere, schließe ich mich evtl. aus der Installation aus oder setze mir ein veraltetes Passwort.

(Da wäre es vlt. geschickter, beim Import zu fragen, ob die User-Tabelle importiert werden darf.)

Reply:
Ich blicke jetzt nicht mehr durch :)

Aktuell schließt der Code doch nun rex_user immer aus, wenn ich das richtig sehe, oder?
Mein Zitat mit deinem +1 sagt doch, dass rex_user stattdessen ggf. doch immer mit exportiert werden soll.
Also müsstest du dafür doch was ändern?

Wobei ich eher dann zu meinem anderen Vorschlag tendiere, rex_user einfach default in die Blacklist zu setzen, und da kann man sie dann rausnehmen.

> (Da wäre es vlt. geschickter, beim Import zu fragen, ob die User-Tabelle importiert werden darf.)

Wobei das nicht so leicht wäre, da ja einfach die sql-Datei importiert wird, und man da nicht so leicht einzelne Tabellen rausnehmen kann.

Reply:
Wir reden ja gerade von 2 Baustellen: Dem manuellen Export und dem Cronjob-Export.

Ich bin kurz beim manuellen Export/Import, wo rex_user ja nicht ausgeschlossen wird, aber demarkiert: Man kann ja auch von Hand die SQL-Datei herunterladen, anpassen und hochladen - wobei man das auch so immer kann. Also eigentlich ist es dann ghupft wia gsprunga (egal).

Man könnte so eine Blacklist natürlich auch auf Backup-Addon-Ebene anlegen, statt rex_user so fest im Code auszuschließen.

Ich bin jetzt auch verwirrt.

Reply:
@gharlan und ich haben gerade dazu telefoniert, ein Update des PRs folgt...